web/src/pages/Configure.tsx

```diff
@@ -0,0 +1,1075 @@
+  name,
+  onClick,
+}: {
+  name: string;
+  onClick: () => void;
+}) => {
+  const { colorMode } = useColorMode();
+
+  return (
+    <GridItem
+      key={name}
+      bg={colorMode === "light" ? "#ECE8FD" : "#2F206E"}
+      fontSize="xs"
+      color={colorMode === "light" ? "#553ACF" : "#ECE8FD"}
+      textOverflow="ellipsis"
+      whiteSpace="nowrap"
+      overflow="hidden"
+      borderRadius="md"
+      px={2}
+      py={1}
+      textAlign="center"
+      _hover={{
+        fontWeight: "bold",
+      }}
+      cursor="pointer"
+      onClick={onClick}
+    >
+      {name}
+    </GridItem>
+  );
+};
+
+const ConfigHeader = ({
+  configInitialized,
+}: {
+  configInitialized: boolean;
+}) => {
+  const [database, setDatabase] = useRecoilState(connectionDatabase);
+  const { colorMode } = useColorMode();
+
+  if (configInitialized) {
+    return <></>;
+  }
+
+  return (
+    <HStack alignItems="flex-end">
+      <FormControl flex={1}>
+        <FormLabel fontSize="xs" fontWeight="bold" textTransform="uppercase">
+          Database name
+        </FormLabel>
+        <Input
+          placeholder="martech"
+          value={database}
+          size="sm"
+          onChange={(e) => setDatabase(e.target.value)}
+        />
+      </FormControl>
+      <Box flex={1}>
+        <ResetSchemaButton
+          background={colorMode === "light" ? "#ECE8FD" : "#2F206E"}
+          color={colorMode === "light" ? "#553ACF" : "#ECE8FD"}
+          size="sm"
+          skipSeedData
+        >
+          Setup schema
+        </ResetSchemaButton>
+      </Box>
+    </HStack>
+  );
+};
+
+const SchemaSection = ({
+  initialized,
+  previousStepCompleted,
+}: {
+  initialized: boolean;
+  previousStepCompleted: boolean;
+}) => {
+  const schemaObjs = useSchemaObjects();
+  const [selectedSchemaObj, setSelectedSchemaObj] = useState<null | string>();
+
+  return (
+    <>
+      {!!selectedSchemaObj && (
+        <SchemaObjectModal
+          onClose={() => setSelectedSchemaObj(null)}
+          schemaObjectName={selectedSchemaObj}
+        />
+      )}
+      <Section
+        completed={initialized}
+        previousStepCompleted={previousStepCompleted}
+        title="Setup the schema"
+        left={
+          <>
+            <Text>
+              A schema includes a database, tables and views to store all the
+              data. Use the tags to create the schema.
+            </Text>
+            <ConfigHeader configInitialized={initialized} />
+          </>
+        }
+        right={
+          <Flex direction="column" gap={3}>
+            <Heading size="sm">Tags</Heading>
+            <SimpleGrid columns={[1, 3, 3]} gap={1}>
+              {Object.keys(schemaObjs.data || {})
+                .sort()
+                .map((name) => (
+                  <SchemaItem
+                    key={name}
+                    name={name}
+                    onClick={() => setSelectedSchemaObj(name)}
+                  />
+                ))}
```

Comment:
can you validate if `schemaObjs.data` exists before so we can remove `|| {} `?

Reply:
Done !